crypto/keys/src/ed25519.rs

```diff
@@ -86,18 +99,20 @@ impl DistinguishedEncoding for Ed25519Signature {
     }
 
     /// Serializes this object into a DER-encoded SubjectPublicKeyInfo structure
-    fn to_der(&self) -> Vec<u8> {
+    fn to_der_slice<'a>(&self, buff: &'a mut[u8]) -> &'a [u8] {
         let data = self.to_bytes();
-        let mut retval = vec![0u8; ED25519_SIG_DER_LEN];
+        buff[..ED25519_SIG_DER_LEN].iter_mut().for_each(|b| *b = 0 );
```

Comment:
If `buff` is too small wouldn't this panic?

Reply:
yep passing a buffer of less than `DistinguishedEncoding::der_size()` will cause a panic, could be an error case but either way it's misuse of the API that _should_ get picked up in the tests

Reply:
@eranrund are y'all okay with this or do i need to propagate the errors through?

(i can't vary the size in the API because that becomes non-object-safe, though i guess i could use `&[u8; DER_MAX_LEN]`)